### PR TITLE
DDF-2933 Resolves issue with dereferenced null value and modification of unmod list

### PR DESCRIPTION
**Pull request by coyotesqrl**
#### What does this PR do?
Fixes issue noted in Coverity scan, CID 161297

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@gordocanchola @harrison-tarr 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@shaundmorris
@stustison

#### How should this be tested? (List steps with links to updated documentation)
Full build and test. For any manual testing, please ask @gordocanchola  for details.

#### Any background context you want to provide?
N/A

#### What are the relevant tickets?
[DDF-2933](https://codice.atlassian.net/browse/DDF-2933)

#### Screenshots (if appropriate)
N/A

#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by harrison-tarr** [comment]
LGTM, I like that it simplifies some of that convoluted `List` logic

File: catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerDelegate.java [hunk 1]
```diff
@@ -348,7 +359,8 @@ public void error(SAXParseException exception) throws SAXException {
     @Override
     public void fatalError(SAXParseException exception) throws SAXException {
         String message = "Fatal Error: " + getParseExceptionInfo(exception);
```
Comment (gordocanchola) [inline catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerDelegate.java]: Should message be broken up and outWriter.appended as well?

Also just curious if you manually made these changes or if it was some formatter suggestions.

Comment (coyotesqrl) [inline catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerDelegate.java]: Ah yes. Intellij caught them as code smells - concats inside appends - and I had it fix them for me. I could certainly replace that concat as well. Give me a sec.

Comment (coyotesqrl) [inline catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerDelegate.java]: Actually...I'm going to leave that one. We pump it to the `outWriter` but also use it as the message on the exception.

**Comment by coyotesqrl** [comment]
build now

File: catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerDelegate.java (hunk 1 above)
Comment (gordocanchola) [inline catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerDelegate.java]: Oh, right. I overlooked that part.

**Pull request merged**